pgx-pg-sys/build.rs

```diff
@@ -536,14 +536,26 @@ fn run_bindgen(pg_config: &PgConfig, include_h: &PathBuf) -> eyre::Result<syn::F
         .header(include_h.display().to_string())
         .clang_arg(&format!("-I{}", includedir_server.display()))
         .parse_callbacks(Box::new(PgxOverrides::default()))
+        .allowlist_file(".*confdefs.h")
+        .allowlist_file(".*([fF]uncs|mgr).h")
+        .allowlist_file(".*htup(_details)?.h")
+        .allowlist_file(".*syscache.h")
+        .allowlist_file(".*mcxt.h")
+        .allowlist_file(".*(storage|catalog|access|commands|executor|adt|optimizer|rewrite|postmaster|tcop|replication|nodes|postgres|parse|pg_|item|heap).*")
+        .allowlist_function("memcpy")
+        .allowlist_var("SIG.*")
         .blocklist_type("Datum") // manually wrapping datum types for correctness
         .blocklist_type("NullableDatum")
         .blocklist_function("varsize_any") // pgx converts the VARSIZE_ANY macro, so we don't want to also have this function, which is in heaptuple.c
         .blocklist_function("query_tree_walker")
         .blocklist_function("expression_tree_walker")
-        .blocklist_function("sigsetjmp")
-        .blocklist_function("siglongjmp")
+        .blocklist_function(".*setjmp")
+        .blocklist_function(".*longjmp")
         .blocklist_function("pg_re_throw")
+        .blocklist_item("CONFIGURE_ARGS") // configuration during build is hopefully irrelevant
+        .blocklist_item("_*(HAVE|have)_.*") // these are for C's use
```

Comment:
These are basically metadata that the CPP logic uses to track which headers have been included and all that. I would like to add all of the `# define HEADER_NAME_H 1` cases as well, which are much the same, but the regex for covering all of those feels less obvious so I didn't take a swing at it.